lib/lte_link_control/lte_lc.c

```diff
@@ -1088,10 +1088,10 @@ int lte_lc_nw_reg_status_get(enum lte_lc_nw_reg_status *status)
 
 	if (!is_cellid_valid(cell_id)) {
 		*status = LTE_LC_NW_REG_UNKNOWN;
+	} else {
```

Comment:
Isn't the commit message incorrect? Doesn't this fix affect the case when the cell ID is actually invalid? Earlier `LTE_LC_NW_REG_UNKNOWN` was overwritten by `status_tmp` when cell ID was not valid.

Reply:
This just adds another branch for `is_cellid_valid() == true`. I don't think it does anything if the cell ID is actually invalid. Where does this overwriting happen? Currently, we just discard `status_tmp`.

Reply:
The original implementation was like this:

```
	if (!is_cellid_valid(cell_id)) {
		*status = LTE_LC_NW_REG_UNKNOWN;
	}

	*status = status_tmp;
```

So, the behavior was incorrect when cell ID was not valid. In that case `status` was first set to `LTE_LC_NW_REG_UNKNOWN`, but immediately overwritten by the value in `status_tmp` (which may have been different).

Reply:
Now I understand. Thanks for pointing that out. I'll update the commit message.